pkg/sql/row/fetcher_mvcc_test.go

```diff
@@ -1,212 +0,0 @@
-	); err != nil {
-		t.Fatal(err)
-	}
-	type rowWithMVCCMetadata struct {
-		PrimaryKey      []string
-		RowIsDeleted    bool
-		RowLastModified string
-	}
-	kvsToRows := func(kvs []roachpb.KeyValue) []rowWithMVCCMetadata {
-		t.Helper()
-		for _, kv := range kvs {
-			log.Infof(ctx, "%v %v %v", kv.Key, kv.Value.Timestamp, kv.Value.PrettyPrint())
-		}
-
-		if err := rf.StartScanFrom(ctx, &row.SpanKVFetcher{KVs: kvs}); err != nil {
-			t.Fatal(err)
-		}
-		var rows []rowWithMVCCMetadata
-		for {
-			datums, _, _, err := rf.NextRowDecoded(ctx)
-			if err != nil {
-				t.Fatal(err)
-			}
-			if datums == nil {
-				break
-			}
-			row := rowWithMVCCMetadata{
-				RowIsDeleted:    rf.RowIsDeleted(),
-				RowLastModified: tree.TimestampToDecimalDatum(rf.RowLastModified()).String(),
-			}
-			for _, datum := range datums {
-				if datum == tree.DNull {
-					row.PrimaryKey = append(row.PrimaryKey, `NULL`)
-				} else {
-					row.PrimaryKey = append(row.PrimaryKey, string(*datum.(*tree.DString)))
-				}
-			}
-			rows = append(rows, row)
-		}
-		return rows
-	}
-
-	var ts1 string
-	sqlDB.QueryRow(t, `BEGIN;
-		INSERT INTO parent VALUES ('1', 'a', 'a', 'a'), ('2', 'b', 'b', 'b');
-		INSERT INTO child VALUES ('1', '10'), ('2', '20');
-		SELECT cluster_logical_timestamp();
-	END;`).Scan(&ts1)
-
-	if actual, expected := kvsToRows(slurpUserDataKVs(t, store.Engine())), []rowWithMVCCMetadata{
-		{[]string{`1`, `a`, `a`, `a`}, false, ts1},
-		{[]string{`1`, `10`}, false, ts1},
-		{[]string{`2`, `b`, `b`, `b`}, false, ts1},
-		{[]string{`2`, `20`}, false, ts1},
-	}; !reflect.DeepEqual(expected, actual) {
-		t.Errorf(`expected %v got %v`, expected, actual)
-	}
-
-	var ts2 string
-	sqlDB.QueryRow(t, `BEGIN;
-		UPDATE parent SET b = NULL, c = NULL, d = NULL WHERE a = '1';
-		UPDATE parent SET d = NULL WHERE a = '2';
-		UPDATE child SET f = '21' WHERE e = '2';
-		SELECT cluster_logical_timestamp();
-	END;`).Scan(&ts2)
-
-	if actual, expected := kvsToRows(slurpUserDataKVs(t, store.Engine())), []rowWithMVCCMetadata{
-		{[]string{`1`, `NULL`, `NULL`, `NULL`}, false, ts2},
-		{[]string{`1`, `10`}, false, ts1},
-		{[]string{`2`, `b`, `b`, `NULL`}, false, ts2},
-		{[]string{`2`, `20`}, true, ts2},
-		{[]string{`2`, `21`}, false, ts2},
-	}; !reflect.DeepEqual(expected, actual) {
-		t.Errorf(`expected %v got %v`, expected, actual)
-	}
-
-	var ts3 string
-	sqlDB.QueryRow(t, `BEGIN;
-		DELETE FROM parent WHERE a = '1';
-		DELETE FROM child WHERE e = '2';
-		SELECT cluster_logical_timestamp();
-	END;`).Scan(&ts3)
-	if actual, expected := kvsToRows(slurpUserDataKVs(t, store.Engine())), []rowWithMVCCMetadata{
-		{[]string{`1`, `NULL`, `NULL`, `NULL`}, true, ts3},
-		{[]string{`1`, `10`}, false, ts1},
-		{[]string{`2`, `b`, `b`, `NULL`}, false, ts2},
-		{[]string{`2`, `20`}, true, ts2}, // ignore me: artifact of how the test is written
-		{[]string{`2`, `21`}, true, ts3},
-	}; !reflect.DeepEqual(expected, actual) {
-		t.Errorf(`expected %v got %v`, expected, actual)
-	}
-}
```

Comment:
It seems (at first glance at least) that this test should remain, probably just with one table instead of two. It seems to me that it exercises more than just interleave-related functionality.